dashboard/app/models/channel_token.rb

```diff
@@ -55,8 +56,13 @@ def self.find_or_create_channel_token(level, ip, user_storage_id, script_id = ni
     end
   end
 
-  def self.find_channel_token(level, user_storage_id)
-    find_by(level: level.host_level, storage_id: user_storage_id)
+  # Finds the channel token. If a channel token exists for the user and level with and without a script ID,
+  # the channel token with the script_id takes precedence.
+  # @param [Level] level The level associated with the channel token request.
+  # @param [String] user_storage_id The if of the storage app associated with the channel token request.
+  # @param [Integer] script_id The ID of the script associated with the channel token.
+  def self.find_channel_token(level, user_storage_id, script_id)
+    order(script_id: 'desc').find_by(level: level.host_level, storage_id: user_storage_id, script_id: [nil, script_id])
```

Comment:
Oh hey, the logic for storing source code against the template level is here (in `level.host_level`).

Reply:
Oh yeah! Now that I think about it this makes sense that the channel token is for the template level, because that's how the source code gets shared between the levels 🤦‍♀️